### PR TITLE
Encoded DeepGPs.

### PR DESCRIPTION
**Pull request by avullo**
**Related issue(s)/PRs:** 

## Summary

Add support for encoding to Deep GPs models.

**Fully backwards compatible:** yes

## PR checklist

- [x] The quality checks are all passing
- [ ] The bug case / new feature is covered by tests
- [ ] Any new features are well-documented (in docstrings or notebooks)

### CODE REVIEW

File: trieste/models/gpflux/sampler.py [hunk 1]
```diff
@@ -96,7 +97,9 @@ def sample(self, at: TensorType, *, jitter: float = DEFAULTS.JITTER) -> TensorTy
         tf.debugging.assert_shapes([(at, [..., 1, None])])
         tf.debugging.assert_greater_equal(jitter, 0.0)
 
-        samples = tf.repeat(at[..., None, :, :], self._sample_size, axis=-3)  # [..., S, 1, D]
+        samples = tf.repeat(
+            self._encode(at[..., None, :, :]), self._sample_size, axis=-3
```
Comment (uri-granta) [inline trieste/models/gpflux/sampler.py:101]: any reason not to encode first thing?

```python
at = self._encode(at)

samples = tf.repeat(at[..., None, :, :], self._sample_size, axis=-3)  # [..., S, 1, D]
```

File: trieste/models/gpflux/sampler.py [hunk 2]
```diff
@@ -72,6 +72,7 @@ def __init__(self, sample_size: int, model: GPfluxPredictor):
             )
             for _ in range(len(self._model_gpflux.f_layers))
         ]
+        self._encode = lambda x: model.encode(x)
```
Comment (uri-granta) [inline trieste/models/gpflux/sampler.py:75]: out of curiosity, does the following work?

```suggestion
        self._encode = model.encode
```

**Review by uri-granta — APPROVED** [review]
LGTM! Couple of very minor questions

File: trieste/models/gpflux/sampler.py (hunk 1 above)
Comment (avullo) [reply trieste/models/gpflux/sampler.py:101]: Mainly compactness.

File: trieste/models/gpflux/sampler.py (hunk 2 above)
Comment (avullo) [reply trieste/models/gpflux/sampler.py:75]: Good point. I think it does.

**Pull request merged**